### ISSUE TITLE
Idea: proptest fallback when not cfg(rmc)

### ISSUE THREAD

**Issue by tedinski** [model-checking/kani#629]
Prerequisite: first, we implement #464 `#[proof]` annotation. (And possibly also have our dev-dependency `rmc` crate in `crates.io` globally.)

Next, besides checking all proofs under `cargo rmc`, we could also (perhaps if an optional feature is enable, like `cfg(rmc-proptest)`) translate down to `proptest` when **not** running under RMC.

The proptest module has things like `prop_assume`: (https://docs.rs/proptest/1.0.0/proptest/#modules) and `assert` works normally. To support this feature, we might need a restricted case where all the `nondet` use is up-front and not dynamic during execution.

This would:

1. Make all proof harnesses "meaningful" and enabled by default in e.g. vscode. (Without having to configure it to know about `cfg(rmc)`)
2. Quickly spot any problems easily spotted by an equivalent property test, not needing full proof. (Notably, this might speed up proof harness development, as wrong proof harnesses would get presumably quicker feedback from the proptest rather than going through RMC.)
3. Make `cargo test` usefully check *something* about RMC harnesses by default, without having to install/run RMC.

**Comment by camshaft** [comment]
I like the idea but I'm wondering if something like this really belongs in this project. The main job of rmc, in my mind, is to compile and execute rust code symbolically. If the project stays at this level, higher level front-ends can integrate and handle the UI. Like you mentioned: 

> To support this feature, we might need a restricted case where all the nondet use is up-front and not dynamic during execution.

This is exactly the job of a front-end: determine useful patterns for writing harnesses and making workflows and interfaces that make it easy to do that. In other words, instead of rmc using proptest, why not have proptest use rmc and handle the switching between the two? This is exactly what we've done in the [bolero integration](https://github.com/camshaft/bolero/tree/rmc-integration) and I think it works quite well. I'm obviously biased in this opinion, but I think it allows for rmc to stay useful for a wide range of target frameworks, each with their own tradeoffs and design decisions.

TL;DR: my vote would be to invest in making it easy to compile and execute rust code symbolically and leave the interface decisions to higher-level frameworks.
